File: support_tools/swigwin-3.0.12/Examples/test-suite/lua/li_factory_runme.lua
require("import")  -- the import fn
import("li_factory") -- import code

-- moving to global
for k,v in pairs(li_factory) do _G[k]=v end

-- catch "undefined" global variables
setmetatable(getfenv(),{__index=function (t,i) error("undefined global variable `"..i.."'",2) end})

circle = Geometry_create(Geometry_CIRCLE)
r = circle:radius()
assert(r == 1.5)

point = Geometry_create(Geometry_POINT)
w = point:width()
assert(w == 1.0)

File: support_tools/swigwin-3.0.12/Examples/test-suite/php/li_factory_runme.php
<?php

require "tests.php";
require "li_factory.php";

// No new functions
check::functions(array(geometry_draw,geometry_create,geometry_clone_,point_draw,point_width,point_clone_,circle_draw,circle_radius,circle_clone_));
// No new classes
check::classes(array(Geometry,Point,Circle));
// now new vars
check::globals(array());

$circle = Geometry::create(Geometry::CIRCLE);
$r = $circle->radius();
check::equal($r, 1.5, "r failed");

$point = Geometry::create(Geometry::POINT);
$w = $point->width();
check::equal($w, 1.0, "w failed");

check::done();
?>
